cmd/bosun/conf/conf.go

```diff
@@ -1,120 +1,120 @@
 package conf // import "bosun.org/cmd/bosun/conf"
 
 import (
-	"encoding/json"
+	"bytes"
 	"fmt"
-	htemplate "html/template"
-	"io/ioutil"
-	"net"
-	"net/http"
+	"hash/fnv"
 	"net/mail"
 	"net/url"
-	"os"
+	"os/exec"
 	"regexp"
-	"runtime"
-	"strconv"
 	"strings"
-	ttemplate "text/template"
 	"time"
 
-	"bosun.org/cmd/bosun/conf/parse"
+	"github.com/influxdata/influxdb/client"
+
 	"bosun.org/cmd/bosun/expr"
-	eparse "bosun.org/cmd/bosun/expr/parse"
+	"bosun.org/cmd/bosun/expr/parse"
 	"bosun.org/graphite"
 	"bosun.org/models"
 	"bosun.org/opentsdb"
+
+	htemplate "html/template"
+	ttemplate "text/template"
+
 	"bosun.org/slog"
-	"github.com/MiniProfiler/go/miniprofiler"
-	"github.com/influxdata/influxdb/client"
 )
 
-type Conf struct {
-	Vars
-	Name            string        // Config file name
-	CheckFrequency  time.Duration // Time between alert checks: 5m
-	DefaultRunEvery int           // Default number of check intervals to run each alert: 1
-	HTTPListen      string        // Web server listen address: :80
-	Hostname        string
-	RelayListen     string // OpenTSDB relay listen address: :4242
-	SMTPHost        string // SMTP address: ny-mail:25
-	SMTPUsername    string // SMTP username
-	SMTPPassword    string // SMTP password
-	Ping            bool
-	PingDuration    time.Duration // Duration from now to stop pinging hosts based on time since the host tag was touched
-	EmailFrom       string
-	StateFile       string
-	LedisDir        string
-	LedisBindAddr   string
+type SystemConfProvider interface {
```

Comment:
I think the interface adds complexity here we don't really need. Why not just a struct without Getter Functions? Are we afraid some code will change it without going through the conf package? I'm not really super worried about that.